src/streaming/ManifestLoader.js

```diff
@@ -194,6 +194,16 @@ function ManifestLoader(config) {
                         logger.debug('BaseURI set by Location to: ' + baseUri);
                     }
 
+                    // If there is a mismatch between the manifest's specified duration and the total duration of all periods,
+                    // overwrite the manifest's duration attribute. This is a patch for if a manifest is generated incorrectly.
+                    if (manifest.mediaPresentationDuration && manifest.Period.length > 1) {
+                        const sumPeriodDurations = manifest.Period.reduce((totalDuration, period) => totalDuration + period.duration, 0);
+                        if (manifest.mediaPresentationDuration - sumPeriodDurations) {
```

Comment:
We should only apply this fix if the duration of all periods is smaller than the mediaPresentationDuration. In case the period duration exceeds the mediaPresentationDuration we should not adjust the mediaPresentationDuration. The author of the MPD might want to play the stream only until mediaPresentationDuration.

Reply:
Makes sense, have updated